FE 309 list scenes scroll

frankyma opened the pull request:
Fixed the sizing of the scene container and only scroll the list of items.  Added a `div` around the list of items

<img width="1680" alt="Screen Shot 2020-07-07 at 6 17 15 PM" src="https://user-images.githubusercontent.com/43487134/86862469-1c36ff00-c07e-11ea-9c38-0112cf981fcd.png">

**Testing**
Open any list scene with items that overflows the page.  The items themselves scroll but not the headers.

CarsonF reviewed (dismissed):
Padding looks weird both for scroll bar and for bottom of list being cut off.
![Screen Shot 2020-07-09 at 6 09 24 PM](https://user-images.githubusercontent.com/932566/87099498-99de4480-c20f-11ea-880e-f1dcf341d664.png)

frankyma commented:
Removed the bottom and right padding on the outside container.  Kept the inner list padding to show card shadow.

CarsonF commented:
Closing for now since no work is being done. Reopen if you make changes.

frankyma commented:
@CarsonF I thought this was pending review?  I don't remember change requests

CarsonF commented:
Sorry I thought we talked about it on slack/zoom. The scroll bar was spaced away from the edge of the window if I remember correctly. I think there might have been something else with the scrolling and bottom of the window, but you might have fixed it.

The project details scene is styled correctly if you want to match that.

frankyma commented:
I remember where we left off.  I fixed the scroll issues but couldn't align the cards, I asked you for some help but you were busy at the time.  I fixed the alignment now with a simple negative margin.

Here's the final view:
![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/43487134/92526845-f7c0e580-f1da-11ea-8dbe-694c7d1a3136.gif)


By the way this isn't styled the same as project details because the list item cards themselves scroll. It's one container level deeper than the detail scenes.  For this case I'm using overflow: 'hidden' on the parent container and overflow: 'auto' on the child list container.

frankyma commented:
Also changed all the overfills to 'auto' from 'scroll' to avoid this (this is on v3 remote)
![Screen Shot 2020-09-08 at 3 45 48 PM](https://user-images.githubusercontent.com/43487134/92534821-a7518400-f1ea-11ea-8b2a-59533f5d0059.png)

glassblowerscat commented on src/components/ContentContainer/ContentContainer.tsx:
```diff
@@ -1,12 +1,13 @@
-    maxWidth: theme.breakpoints.values.md,
+    overflow: 'hidden',
+    padding: spacing(4, 0, 0, 4),
+    display: 'flex',
+    flexDirection: 'column',
```
This seems effective for the problem we're trying to solve, but I think we need to add some styling to `ProjectFilesList` to counteract this:

![image](https://user-images.githubusercontent.com/728777/93416689-440fc380-f874-11ea-9ef0-0eb1bd84ebb0.png)

glassblowerscat replied:
It's supposed to look like this:

![image](https://user-images.githubusercontent.com/728777/93416747-63a6ec00-f874-11ea-9a20-62a49a6445d1.png)

glassblowerscat replied:
Could just be as simple as wrapping the table in a `Grid`

frankyma replied:
I just gave the table some padding on the right and bottom
![Screen Shot 2020-09-18 at 1 54 21 PM (2)](https://user-images.githubusercontent.com/43487134/93644343-a53fb000-f9b6-11ea-8f62-aa2d19c1e226.png).  Don't think we need Grid for this since it's just a single item.  Let me know what you think

glassblowerscat replied:
What I was thinking with the `Grid` was that we ought to keep it at that 2/3-ish width it was at before... thought I remembered that being something @CarsonF and I discussed, but maybe not. Care to weigh in, Carson?

glassblowerscat commented on src/scenes/Languages/List/LanguageList.tsx:
```diff
@@ -14,6 +14,14 @@ const useStyles = makeStyles(({ spacing }) => ({
   options: {
     margin: spacing(3, 0),
   },
+  listContainer: {
+    overflow: 'auto',
+    marginLeft: spacing(-2),
+    padding: spacing(2),
+  },
+  item: {
+    marginBottom: spacing(2),
+  },
```
We repeat this a lot. Can we break this out into a simple `ListScrollContainer` or similar? It might be worth moving that new component and `ContentContainer` into a new `components/Layout` directory; what do you think?

frankyma replied:
good idea, implemented.
